### DISCUSSION TITLE
11.12.2023 - Programming Languages

### DISCUSSION THREAD

**Discussion by DGrillCoder** [grilldb97/legendary-space-zebra#10] [Show and tell]
Heute haben wir uns in der Programmierung mit Sprachen in der Programmierung beschäftigt.
Dafür haben wir uns das wissenschaftliche Paper mit dem Titel “Energy Efficiency across Programming Languages” von der Universität Minho](https://greenlab.di.uminho.pt/wp-content/uploads/2017/10/sleFinal.pdf) angeschaut. Dieses untersucht den Energie- und Speicherverbrauch, sowie die Laufzeit von 27 bekannten Programmiersprachen. Man ist zum Ergebnis gekommen, dass langsamere Sprachen weniger Energie verbrauchen, als schnellere.
In Gruppen haben wir uns mit der wissenschaftlichen Arbeit zusammengetan und uns mit unterschiedlichen Fragen dazu auseinander gesetzt (z.B. Was ist Energie?; Was sind Benchmarks?; War ist Paradigmen?,...)

Meine Gruppe hat AI genutzt, um die Fragen in Bezug auf des Textes zu beantworten.
Zuerst haben wir den Text lesen lassen und anschließend Fragen gestellt.
Das hat soweit gut geklappt: Es waren sinnvolle und verständliche Antworten und Zusammenfassungen des wissenschaftlichen Textes dabei.  Wir haben lediglich nur einmal darauf hinweisen müssen, dass die AI den Bezug auf den Text nehmen soll.

Python ist einer der Sprachen, welche am meisten Energie benötigt. Dies liegt daran, dass diese Programmiersprache interpretiert ist und der Code zur Laufzeit ausgeführt wird. Somit kann es dazu führen, dass Python langsamer ist als kompilierte Sprachen wie Java und somit mehr Energie benötigt. Auch die dynamische Typisierung (Typ-Prüfungen in der Laufzeit, anstatt bei der Kompilierung) trägt bzgl. seines Speicherverbrauchs zu einem höheren Energieverbrauch bei. Durch die Abstraktion (Vereinfachung komplexer Systeme) in Python wird die Entwicklung erleichtert, aber die Hardware-Ressourcen könnten ggf. nicht optimal genutzt werden, was wiederum die Effizienz verringert.

Die Effizienz der Programmiersprachen hängt von vielen verschiedenen Faktoren ab und es muss abgewägt werden, welche Sprache für den Einsatz am Besten ist.

Im Großen und Ganzen ist es ein spannendes Thema, welches man nicht aus den Augen verlieren sollte. Auch sollte man das Thema Effizient der Programmiersprachen in Projekten nicht vernachlässigen.

**Comment by SirPatschiii** [comment]
Saubere & informative Zusammenfassung.
Definitiv ein interessantes Themengebiet, insbesondere wenn man das alles im großen Stil betrachtet.

**Comment by Roadraudi** [comment]
Oh sehr cool. Vor allem auch KI einzubauen und um einem Arbeit zu ersparen.

**Comment by CookieClickr** [comment]
Gut gemacht, weiter so!
